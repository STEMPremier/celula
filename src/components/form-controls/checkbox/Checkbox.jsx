/* eslint-disable no-console */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import './checkbox.less';

class Checkbox extends Component {
  state = {
    checked: false,
    showFocus: false,
  };

  componentDidMount() {
    const { checked } = this.props;

    if (checked) {
      this.setState({
        checked: true,
      });
    }
  }

  handleChange = event => {
    const { handleChange } = this.props;

    this.setState({
      checked: event.target.checked,
    });

    handleChange(event.target.value);
  };

  onMouseEnter = () => {
    this.setState(() => ({
      showFocus: true,
    }));
  };

  onMouseLeave = () => {
    this.setState(() => ({ showFocus: false }));
  };

  render() {
    const {
      className,
      disabled,
      errorMsg,
      formId,
      name,
      label,
      value,
    } = this.props;
    const { checked, showFocus } = this.state;
    const id = `${name}_${value}`;
    const classes = cx(
      'ce-checkbox',
      {
        'ce-checkbox--disabled': disabled,
        'ce-checkbox--individual-checkbox-error': errorMsg,
      },
      className,
    );

    return (
      <div
        className={classes}
        onMouseEnter={this.onMouseEnter}
        onMouseLeave={this.onMouseLeave}
      >
        <div className="ce-checkbox--box">
          <input
            checked={checked}
            disabled={disabled}
            form={formId}
            id={id}
            name={name}
            onChange={this.handleChange}
            type="checkbox"
            value={value}
          />
          <label htmlFor={id}>{label}</label>
          <div className="ce-checkbox--background" />
          {showFocus && <div className="ce-checkbox--focus" />}
          {errorMsg && (
            <div className="ce-checkbox--error-box-wrapper">
              <div className="ce-checkbox--arrow" />
              <div className="ce-checkbox--error-box">
                <div className="ce-checkbox--error-box-text">{errorMsg}</div>
              </div>
            </div>
          )}
        </div>
      </div>
    );
  }
}

Checkbox.propTypes = {
  /**
   * Select the `<Checkbox />`.
   */
  checked: PropTypes.bool,
  /**
   * A class name added to the `<CheckboxGroup />`.
   */
  className: PropTypes.string,
  /**
   * Make the `<Checkbox />` inactive.
   */
  disabled: PropTypes.bool,
  /**
   * An error message to display for a `<Checkbox />'.
   */
  errorMsg: PropTypes.string,
  /**
   * The id of the form the `<Checkbox />` belongs to. This value is overwritten by any value passed from an enclosing `<CheckboxGroup />`.
   */
  formId: PropTypes.string,
  /**
   * A function to trigger when the state of the `<Checkbox />` changes.
   */
  handleChange: PropTypes.func,
  /**
   * This component is to be treated as a checkbox in `<FormControlGroup />`.
   * @ignore
   */
  isA: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
  /**
   * The `<Checkbox />` label.
   */
  label: PropTypes.string.isRequired,
  /**
   * Need to assign a matching name to each `<Checkbox />` if it is being used outside of the `<CheckboxGroup />`.  In  a `<CheckboxGroup />`, each individual `<Checkbox />` will have the same name, but that will come from the `<CheckboxGroup />`.
   */
  name: PropTypes.string,
  /**
   * The value of the `<Checkbox />`.
   */
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]).isRequired,
};

Checkbox.defaultProps = {
  checked: false,
  className: '',
  disabled: false,
  errorMsg: '',
  formId: '',
  handleChange: () => {},
  isA: 'checkbox',
  name: '',
};

export default Checkbox;
