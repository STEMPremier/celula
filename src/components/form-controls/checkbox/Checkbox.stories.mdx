import { Description, Meta, Preview, Props, Story } from '@storybook/addon-docs/blocks';
import { action } from '@storybook/addon-actions';
import { boolean, select, text, array } from '@storybook/addon-knobs';

import { Checkbox } from '.';
import { COMPONENT_STATES as STATES } from '../../../utils/constants';

<Meta component={Checkbox} title="Components/Molecules/Form Controls/Checkbox"/>

# Checkbox
<Description of={Checkbox} />
<Props of={Checkbox}/>

<Story name="Checkbox with Knobs" parameters={{docs: {disable: true}}}>
  <form id="formKnobs">
    <Checkbox
      checked={boolean('checked', false )}
      className={text('className', '')}
      disabled={boolean('disabled', false)}
      errorMsg={text('errorMsg', '')}
      formId={text('formId', 'formKnobs')}
      handleClick={action('`Checkbox with Knobs` onClick event triggered.')}
      label={text('label', 'Accept Terms and Conditions')}
      name={text('name', 'termsAndConditions')}
      value={text('value', 'accepted')}
    />
  </form>
</Story>

### Types
<Preview>
  <Story name="Checkbox Types">
    <form id="formTypes">
      <Checkbox
        checked
        formId="formTypes"
        label="I am checked"
        name="checked"
        value={true}
      />
      <Checkbox
        formId="formTwo"
        label="I am unchecked"
        name="unchecked"
        value={false}
      />
    </form>
  </Story>
</Preview>


### States
<Preview>
  <Story name="Checkbox States">
    <form id="formStates">
      {STATES.map(state => (
        <Checkbox
          disabled={state === 'disabled'}
          errorMsg={state === 'error' ? "There is an error." : ''}
          formId="formStates"
          key={state}
          label={`${state.charAt(0).toUpperCase()}${state.slice(1)} Checkbox`}
          name={`${state}_checkbox`}
          value={state}
        />
      ))}
    </form>
  </Story>
</Preview>
