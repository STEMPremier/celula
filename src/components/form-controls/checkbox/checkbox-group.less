@import '../../../styles/celula';

.ce-checkbox-group {
  border: 0.3rem solid @white;
  border-radius: 0.5rem;
  clear: both;
  margin-bottom: 0.7rem;

  legend {
    margin-bottom: 2.2rem;
    margin-left: 2.2rem;
    padding: 0 0.3rem;
  }

  span {
    background-color: @white;
    display: block;
    float: right;
    height: 1.6rem;
    margin-bottom: -1rem;
    margin-right: 2.2rem;
    margin-top: 2.2rem;
    padding: 0 0.3rem;
  }

  &.ce-checkbox-group--disabled {
    pointer-events: none;

    input {
      + label {
        // Unchecked
        // The background
        &::before {
          background-color: @unchecked-disabled-gray;
        }
      }

      &:checked + label {
        // Checked
        // The background
        &::before {
          background-color: @checked-disabled-gray;
        }

        // The dot
        &::after {
          background-color: @checked-disabled-gray;
        }
      }
    }
  }

  &.ce-checkbox-group--error {
    border: 0.3rem solid @error-text;
    color: @error-text;

    input:checked + label {
      // Checked
      // The background
      &::before {
        background-color: @checked-red;
      }

      // The dot
      &::after {
        background-color: @checked-red;
        border: 0.5rem solid @white;
      }

      // Hover
      &:hover {
        // The background
        &::before {
          background-color: @hover-red;
        }
      }

      // Active
      // The background
      &:active {
        &::before {
          background-color: @active-red;
        }
      }

      // Focus
      &:focus {
        // The background
        &::before {
          background-color: @focus-red;
        }
      }
    }
  }
}
