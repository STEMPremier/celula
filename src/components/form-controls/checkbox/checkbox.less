@import '../../../styles/celula';

.ce-checkbox {
  user-select: none;

  input {
    opacity: 0;

    + label {
      cursor: pointer;
      display: inline-block;
      height: 2rem;
      line-height: 2rem;
      padding: 0.8rem 4.4rem;

      // Unchecked
      // The background
      &::before {
        background-color: transparent;
        border: 0.75rem solid transparent;
        border-radius: 50%;
        box-sizing: border-box;
        content: '';
        height: 3.6rem;
        left: 0;
        margin: -0.8rem;
        margin-left: 2rem;
        position: absolute;
        width: 3.6rem;
      }

      // The dot
      &::after {
        background-color: @unchecked-background;
        border-radius: 0.3rem;
        box-sizing: border-box;
        content: '';
        height: 2rem;
        left: 0;
        margin-left: 2.8rem;
        position: absolute;
        width: 2rem;
      }

      // Hover
      &:hover {
        // The background
        &::before {
          background-color: @hover-gray;
          border: 0;
        }
      }

      // Active
      &:active {
        // The background
        &::before {
          background-color: @active-gray;
          border: 0;
        }
      }

      // Focus
      &:focus {
        // The background
        &::before {
          background-color: @focus-gray;
          border: 0;
        }
      }

      // The background mask
      + .ce-checkbox__center {
        border: 1px solid @unchecked-gray;
        border-radius: 0.3rem;
        box-sizing: border-box;
        display: inline-block;
        height: 2.1rem;
        left: 0;
        margin-left: 2.75rem;
        margin-top: 0.75rem;
        position: absolute;
        width: 2.1rem;
        z-index: -1;
      }
    }

    &:checked + label {
      // Checked
      // The dot
      &::after {
        background-color: @checked-blue;
        border: 0.5rem solid @white;
      }

      // Hover
      &:hover {
        // The background
        &::before {
          background-color: @hover-blue;
        }
      }

      // Active
      // The background
      &:active {
        &::before {
          background-color: @active-blue;
        }
      }

      // Focus
      &:focus {
        // The background
        &::before {
          background-color: @focus-blue;
        }
      }

      // The background mask
      + .ce-checkbox__center {
        border: 1px solid @checked-blue;
      }
    }
  }

  &.ce-checkbox--disabled {
    pointer-events: none;

    input {
      + label {
        // Unchecked
        // The background mask
        + .ce-checkbox__center {
          border: 1px solid @unchecked-disabled-gray;
        }
      }

      &:checked + label {
        // Checked
        // The background
        &::before {
          background-color: @checked-disabled-gray;
        }

        // The dot
        &::after {
          background-color: @checked-disabled-gray;
        }
      }

      // The background mask
      + .ce-checkbox__center {
        border: 1px solid @checked-blue;
      }
    }
  }

  &.ce-checkbox--error {
    input {
      + label {
        color: @error-text;

        // Unchecked
        // The background mask
        + .ce-checkbox__center {
          border: 1px solid @checked-red;
        }
      }

      &:checked + label {
        // Checked
        // The dot
        &::after {
          background-color: @checked-red;
        }

        // Hover
        &:hover {
          // The background
          &::before {
            background-color: @hover-red;
            border: 0;
          }
        }

        // Active
        &:active {
          // The background
          &::before {
            background-color: @active-red;
            border: 0;
          }
        }

        // Focus
        &:focus {
          // The background
          &::before {
            background-color: @focus-red;
            border: 0;
          }
        }

      }

      // The background mask
      + .ce-checkbox__center {
        border: 1px solid @checked-red;
      }
    }

    .ce-checkbox--error__container {
      background-color: @white;
      padding: 0 2.75rem;

      .ce-checkbox--error__arrow {
        border-bottom: 0.5rem solid @error-box-background;
        border-left: 0.5rem solid transparent;
        border-right: 0.5rem solid transparent;
        margin-left: 25%;
        width: 0;
      }

      .ce-checkbox--error__message {
        background-color: @error-box-background;
        border-radius: 0.5rem;
        display: inline-block;
        padding: 2rem 5rem;

        .ce-checkbox--error__text {
          color: @white;
          font-size: @error-font-size;
        }
      }
    }
  }
}

.ce-checkbox + .ce-checkbox {
  margin-top: 0.4rem;
}
