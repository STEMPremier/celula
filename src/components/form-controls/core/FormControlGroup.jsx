/* eslint-disable prettier/prettier */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

class FormControlGroup extends Component {
  state = {
    selectedValues: this.props.selectedValues, // eslint-disable-line react/destructuring-assignment
  };

  handleCheckboxChange = event => {
    const { handleChange } = this.props;
    let { selectedValues } = this.state;

    if (event.target.checked) {
      selectedValues.push(event.target.value);
    } else {
      selectedValues = selectedValues.filter(val => val !== event.target.value);
    }

    handleChange(event.target.value);

    this.setState({ selectedValues });
  };

  handleRadioChange = event => {
    const { handleChange } = this.props;

    handleChange(event.target.value);

    this.setState({ selectedValues: [event.target.value] });
  };

  handleChange = event => {
    let { children } = this.props;
    let childType;

    children = React.Children.toArray(children);

    for ( const child of children ) { // eslint-disable-line no-restricted-syntax
      childType = child.props.isA;
      break;
    }

    // there are only 2 types of children, ever
    if (childType === 'radio') this.handleRadioChange(event);
    if (childType === 'checkbox') this.handleCheckboxChange(event);
  };

  renderChildren = () => {
    const { children, name } = this.props;
    const { selectedValues } = this.state;

    return React.Children.map(children, child => {
      const props = {
        checked: selectedValues.includes(child.props.value),
        name,
      };

      return React.cloneElement(child, { ...props });
    });
  };

  render() {
    const { className, disabled, errorClass, errorMsg, formId, label } = this.props;

    return (
      <fieldset
        className={className}
        disabled={disabled}
        form={formId}
        onChange={this.handleChange}
      >
        <legend>{label}</legend>
        {this.renderChildren()}
        {errorMsg && <div className={errorClass}>{errorMsg}</div>}
      </fieldset>
    );
  }
}

FormControlGroup.propTypes = {
  /**
   * The components you want the `<FormControlGroup />` to group together.
   */
  children: PropTypes.node.isRequired,
  /**
   * A class name, or a string of class names, to add to the `<FormControlGroup />`.
   */
  className: PropTypes.string,
  /**
   * Disables the `<FormControlGroup />` and all of it's children.
   */
  disabled: PropTypes.bool,
  /**
   * A class name, or string of class names, to add to your `<FormControlGroup />` if there is an error message.
   */
  errorClass: PropTypes.string,
  /**
   * An error message to display in the `<FormControlGroup />`.
   */
  errorMsg: PropTypes.string,
  /**
   * The id of the form the `<FormControlGroup />` belongs to.
   */
  formId: PropTypes.string,
  /**
   * A function to trigger when the state of the `<FormControlGroup />` changes.
   */
  handleChange: PropTypes.func,
  /**
   * The `<FormControlGroup />` legend.
   */
  label: PropTypes.string.isRequired,
  /**
   * The name given to all the children of the `<FormControlGroup />`.
   */
  name: PropTypes.string.isRequired,
  /**
   * An array of values used to pre-select children of the `<FormControlGroup />`.
   */
  selectedValues: PropTypes.arrayOf(PropTypes.any),
};

FormControlGroup.defaultProps = {
  className: '',
  disabled: false,
  errorMsg: '',
  errorClass: '',
  formId: '',
  handleChange: () => {},
  selectedValues: [],
};

export default FormControlGroup;
/* eslint-enable prettier/prettier */
