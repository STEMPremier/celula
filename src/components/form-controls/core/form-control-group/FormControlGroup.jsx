/* eslint-disable prettier/prettier */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

/**
 *  `FormControlGroup` is used for grouping arbitrary form-control-like componets that need to operate like checkboxes or a radio buttons.
 *  It is the basis for the 'gouping' components found through out the ui kit.
 *
 * `FormControlGroup` has no specific styles beyone those it children componets bring with them.
 */
const FormControlGroup = (
  {
    children,
    className,
    disabled,
    errorClass,
    errorMsg,
    formId,
    handleChange: handler,
    label,
    name,
    selectedValues,
  },
  ...props
) => {
  const [fieldValues, setFieldValues] = useState(selectedValues);

  useEffect(() => {
    setFieldValues(selectedValues);
  }, [selectedValues]);

  const handleCheckboxChange = event => {
    const { checked, value } = event.target;
    let selectedVals = [...fieldValues];

    if (checked) {
      selectedVals.push(value);
    } else {
      selectedVals = selectedVals.filter(val => val !== value);
    }

    setFieldValues(selectedVals);
    handler(event, selectedVals);
  };

  const handleRadioChange = event => {
    const { value } = event.target;

    setFieldValues([value]);
    handler(event);
  };

  const handleChange = event => {
    const childs = React.Children.toArray(children);
    let childType;

    for ( const child of childs ) { // eslint-disable-line no-restricted-syntax
      childType = child.props.isA;
      break;
    }

    // there are only 2 types of children, ever
    if (childType === 'radio') handleRadioChange(event);
    if (childType === 'checkbox') handleCheckboxChange(event);
  };

  const renderChildren = () => {
    const { childProps } = props;

    return React.Children.map(children, child => {
      const newProps = {
        checked: fieldValues.includes(child.props.value),
        name,
        ...childProps,
      };

      return React.cloneElement(child, { ...newProps });
    });
  };

  return (
    <fieldset
      className={className}
      disabled={disabled}
      form={formId}
      onChange={handleChange}
    >
      <legend>{label}</legend>
      {renderChildren()}
      <span className={errorClass}>{errorMsg}</span>
    </fieldset>
  );
};

FormControlGroup.propTypes = {
  /**
   * Props passed to all the children of the `<FormControlGroup />`.
   */
  childProps: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  /**
   * The components you want the `<FormControlGroup />` to group together.
   */
  children: PropTypes.node.isRequired,
  /**
   * A class name, or a string of class names, to add to the `<FormControlGroup />`.
   */
  className: PropTypes.string,
  /**
   * Disables the `<FormControlGroup />` and all of it's children.
   */
  disabled: PropTypes.bool,
  /**
   * A class name, or string of class names, to add to your `<FormControlGroup />` if there is an error message.
   */
  errorClass: PropTypes.string,
  /**
   * An error message to display in the `<FormControlGroup />`.
   */
  errorMsg: PropTypes.string,
  /**
   * The id of the form the `<FormControlGroup />` belongs to.
   */
  formId: PropTypes.string,
  /**
   * A function to trigger when the state of the `<FormControlGroup />` changes.
   */
  handleChange: PropTypes.func,
  /**
   * The `<FormControlGroup />` legend.
   */
  label: PropTypes.string,
  /**
   * The name given to all the children of the `<FormControlGroup />`.
   */
  name: PropTypes.string.isRequired,
  /**
   * An array of values used to pre-select children of the `<FormControlGroup />`.
   */
  selectedValues: PropTypes.arrayOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  ),
};

FormControlGroup.defaultProps = {
  childProps: {},
  className: '',
  disabled: false,
  errorMsg: '',
  errorClass: '',
  formId: '',
  handleChange: () => {},
  label: '',
  selectedValues: [],
};

export default FormControlGroup;
/* eslint-enable prettier/prettier */
