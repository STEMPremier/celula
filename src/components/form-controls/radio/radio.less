@import '../../../styles/celula';

.ce-radio {
  user-select: none;

  input {
    opacity: 0;

    + label {
      cursor: pointer;
      display: inline-block;
      height: 2rem;
      line-height: 2rem;
      padding: 0.8rem 4.4rem;

      // Unchecked
      // The background
      &::before {
        background-color: @unchecked-gray;
        border: 0.75rem solid @white;
        border-radius: 50%;
        box-sizing: border-box;
        content: '';
        height: 3.6rem;
        left: 0;
        margin: -0.8rem;
        margin-left: 2rem;
        position: absolute;
        width: 3.6rem;
      }

      // The dot
      &::after {
        background-color: @unchecked-background;
        border-radius: 50%;
        box-sizing: border-box;
        content: '';
        height: 2rem;
        left: 0;
        margin-left: 2.8rem;
        position: absolute;
        width: 2rem;
        box-shadow: -0.5px 0 @unchecked-gray,
          0.5px 0 @unchecked-gray,
          0 -0.5px @unchecked-gray,
          0 0.5px @unchecked-gray;
      }

      // Hover
      &:hover {
        // The background
        &::before {
          background-color: @hover-gray;
          border: 0;
        }

        &::after {
          box-shadow: none;
        }
      }

      // Active
      &:active {
        // The background
        &::before {
          background-color: @active-gray;
          border: 0;
        }

        &::after {
          box-shadow: none;
        }
      }

      // Focus
      &:focus {
        // The background
        &::before {
          background-color: @focus-gray;
          border: 0;
        }

        &::after {
          box-shadow: none;
        }
      }
    }

    &:checked + label {
      // Checked
      // The background
      &::before {
        background-color: @checked-blue;
      }

      // The dot
      &::after {
        background-color: @checked-blue;
        border: 0.5rem solid @white;
      }

      // Hover
      &:hover {
        // The background
        &::before {
          background-color: @hover-blue;
        }
      }

      // Active
      // The background
      &:active {
        &::before {
          background-color: @active-blue;
        }
      }

      // Focus
      &:focus {
        // The background
        &::before {
          background-color: @focus-blue;
        }
      }
    }
  }
}

.ce-radio + .ce-radio {
  margin-top: 0.4rem;
}
