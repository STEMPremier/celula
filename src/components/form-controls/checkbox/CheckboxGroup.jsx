import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import FormControlGroup from '../core';

import './checkbox-group.less';

/**
 * `Checkboxes` allow the user to select one or more items from a set. They can be used to turn an option on or off, select one or multiple items from a list, or present a list containing sub-selections.
 */
class CheckboxGroup extends Component {
  renderChildren = () => {
    const { children, name, selectedValues } = this.props;

    return React.Children.map(children, child => {
      const props = {
        selectedValues,
        name,
      };
      return React.cloneElement(child, { ...props });
    });
  };

  render() {
    const {
      children,
      className,
      disabled,
      errorMsg,
      formId,
      handleChange,
      label,
      name,
      selectedValues,
      // validators
    } = this.props;

    const classes = cx(
      'ce-checkbox-group',
      {
        'ce-checkbox-group--disabled': disabled,
        'ce-checkbox-group--error': errorMsg,
      },
      className,
    );

    return (
      <FormControlGroup
        className={classes}
        disabled={disabled}
        errorClass="ce-checkbox-group--error__text"
        errorMsg={errorMsg}
        formId={formId}
        handleChange={handleChange}
        label={label}
        name={name}
        selectedValues={selectedValues}
      >
        {children}
      </FormControlGroup>
    );
  }
}

CheckboxGroup.propTypes = {
  /**
   * The `<Checkbox />` components you want the `<CheckboxGroup />` to group together.
   */
  children: PropTypes.node.isRequired,
  /**
   * A class name, or string of class names, to add to the `<CheckboxGroup />`.
   */
  className: PropTypes.string,
  /**
   * Disables the `<CheckboxGroup />` and all of its children.
   */
  disabled: PropTypes.bool,
  /**
   * An error message to display in the `<CheckboxGroup />`.
   */
  errorMsg: PropTypes.string,
  /**
   * The id of the form the `<CheckboxGroup />` belongs to.
   */
  formId: PropTypes.string,
  /**
   * A function to trigger when the state of the `<CheckboxGroup />` changes.
   */
  handleChange: PropTypes.func,
  /**
   * The `<CheckboxGroup />` legend.
   */
  label: PropTypes.string.isRequired,
  /**
   * The name given to all the children of the `<CheckboxGroup />`.
   */
  name: PropTypes.string.isRequired,
  /**
   * The values used to pre-select some children of the `<CheckboxGroup />`.
   */
  selectedValues: PropTypes.arrayOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  ),
  // validators: PropTypes.array,
};

CheckboxGroup.defaultProps = {
  className: '',
  disabled: false,
  errorMsg: '',
  formId: '',
  handleChange: () => {},
  selectedValues: [],
  // validators: [],
};

export default CheckboxGroup;
