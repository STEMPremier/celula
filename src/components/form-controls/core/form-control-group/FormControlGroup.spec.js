import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FormControlGroup from '.';
import { Radio } from '../../radio';
import { Checkbox } from '../../checkbox';

describe('FormControlGroup', () => {
  describe('render()', () => {
    test('render a FCG with Checkboxes', () => {
      const wrapper = shallow(
        <FormControlGroup
          label="FormControlGroup Test Label"
          name="FormControlGroup Test Name"
          handleChangeGroup={() => {}}
        >
          <Checkbox
            label="Checkbox Test Label 1"
            value="Checkbox Value Here 1"
          />
          <Checkbox
            label="Checkbox Test Label 2"
            value="Checkbox Value Here 2"
          />
        </FormControlGroup>,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('render a FCG with Radios', () => {
      const wrapper = shallow(
        <FormControlGroup
          label="FormControlGroup Test Label"
          name="FormControlGroup Test Name"
          handleChangeGroup={() => {}}
        >
          <Radio label="Radio Test Label 1" value="Radio Value Here 1" />
          <Radio label="Radio Test Label 2" value="Radio Value Here 2" />
        </FormControlGroup>,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
