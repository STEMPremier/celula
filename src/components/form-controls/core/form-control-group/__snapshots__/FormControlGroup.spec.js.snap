// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControlGroup render() render a FCG with Checkboxes 1`] = `
<fieldset
  className=""
  disabled={false}
  form=""
  onChange={[Function]}
  style={Object {}}
>
  <legend>
    FormControlGroup Test Label
  </legend>
  <Checkbox
    checked={false}
    className=""
    disabled={false}
    errorMsg=""
    formId=""
    handleChange={[Function]}
    isA="checkbox"
    key=".0"
    label="Checkbox Test Label 1"
    name="FormControlGroup Test Name"
    style={Object {}}
    value="Checkbox Value Here 1"
  />
  <Checkbox
    checked={false}
    className=""
    disabled={false}
    errorMsg=""
    formId=""
    handleChange={[Function]}
    isA="checkbox"
    key=".1"
    label="Checkbox Test Label 2"
    name="FormControlGroup Test Name"
    style={Object {}}
    value="Checkbox Value Here 2"
  />
  <span
    className=""
  />
</fieldset>
`;

exports[`FormControlGroup render() render a FCG with Radios 1`] = `
<fieldset
  className=""
  disabled={false}
  form=""
  onChange={[Function]}
  style={Object {}}
>
  <legend>
    FormControlGroup Test Label
  </legend>
  <Radio
    checked={false}
    className=""
    disabled={false}
    isA="radio"
    key=".0"
    label="Radio Test Label 1"
    name="FormControlGroup Test Name"
    style={Object {}}
    value="Radio Value Here 1"
  />
  <Radio
    checked={false}
    className=""
    disabled={false}
    isA="radio"
    key=".1"
    label="Radio Test Label 2"
    name="FormControlGroup Test Name"
    style={Object {}}
    value="Radio Value Here 2"
  />
  <span
    className=""
  />
</fieldset>
`;
